common/src/main/java/io/netty/util/internal/MpscLinkedQueue.java

```diff
@@ -248,53 +256,37 @@ public E element() {
         throw new NoSuchElementException();
     }
 
-    @Override
-    public Object[] toArray() {
-        final Object[] array = new Object[size()];
-        final Iterator<E> it = iterator();
-        for (int i = 0; i < array.length; i ++) {
-            if (it.hasNext()) {
-                array[i] = it.next();
-            } else {
-                return Arrays.copyOf(array, i);
+    private List<E> toList(int initialCapacity) {
+        List<E> elements = new ArrayList<E>(initialCapacity);
+
+        MpscLinkedQueueNode<E> n = peekNode();
+        for (;;) {
+            if (n == null) {
+                break;
             }
+            E value = n.value();
+            if (value == null) {
+                break;
+            }
+            elements.add(value);
+            MpscLinkedQueueNode<E> next = n.next();
+            if (n == next) {
+                break;
+            }
+            n = next;
         }
-        return array;
+        return elements;
+    }
+
+    @Override
+    public Object[] toArray() {
+        return toList(16).toArray();
```

Comment:
Agreed not a big deal...but why have magic numbers when you can not have them...

Change this:
```java
private List<E> toList(int initialCapacity) {
..
}
@Override
    public Object[] toArray() {
        return toList(16).toArray();
    }
```

to this:
```java
private List<E> toList(List<E> elements) {
..
}
@Override
    public Object[] toArray() {
        return toList(new ArrayList<E>()).toArray();
    }
```

then also search and replace the other magic `16` number floating around in [this file](https://github.com/netty/netty/pull/3684/files#diff-ac5cb182f74f909912b9cce8a20556aaR214)